### ISSUE TITLE
Allow adding a class to an event while dragging

### ISSUE THREAD

**Issue by sanderdewilde** [fullcalendar/fullcalendar#5996]
### Checklist

Please mark these items with an [x]

  - [x] I've already searched through [existing tickets](https://fullcalendar.io/issues)
  - [x] Other people will find this feature useful

### Feature Description

I'm looking for the possibility to add a class to an event while dragging (timeline view).

In our application, some drop locations are not allowed from business perspective. This is supported by fullcalendar by both the `constraint` mechanism and the `eventAllow` callback.

However, we want the user to be able to drag the event to the non-allowed locations anyway, as the end user must always be able to override the constraints our system thinks are valid. We would then warn the user by rendering the event in a different colour. Therefore, we'd like to add a class to the event. This should happen while dragging the event around (so before the `eventDragStop` / `eventDrop` callbacks).

The behaviour is already present in the library to some extend. E.g., when dragging an event to the past (over the "now" line), the styling of the `fc-event-past` class is applied while dragging. We'd need the exact same thing, but we'd need to be able to assign the class based on our own validation logic. So it's kind of a mix between the `eventAllow` and `eventClassNames` callbacks.

I think the `eventClassNames` callback is executed upon each movement of the event, so adding the `dropInfo` (might need to rename it here) as an extra argument might be sufficient. But I don't know the internals, so I might be dropping the ball here.

I have tried to combine the `eventAllow` and `eventClassNames` callbacks like so:
```js
eventAllow={(dropInfo, draggedEvent) => {
  if (draggedEvent) {
    // we'd check if the drop is allowed here
    draggedEvent.setExtendedProp('forbidden', true);
  }
  return true;
}}
eventClassNames={info => {
  let className = info.event.extendedProps['forbidden'] ? "event-forbidden" : "";
  return className;
}}
```
but the `forbidden` `extendedProp` does not seem to be persisted and the class is not applied.

Furthermore, I've tried setting the `backgroundColor` property on the `draggedEvent` in the `eventAllow` callback, but the property is readonly so this does not work either.

Related question: as in my use case we'd need to implement checking if a drop is allowed ourselves, can the logic that fullcalendar uses be used in a standalone fashion? So we'd want to provide the dropInfo and constraints as an input, and receive the decision as output.

Thanks,
Sander

**Comment by acerix** [comment]
I don't believe it's possible to modify the event while dragging, but another way you could change the style from eventAllow is to change which styles are applied to the "fc-event-mirror" class which is on the event being dragged:

https://codepen.io/acerix/pen/oNzbBKd?editors=0110

I'm not sure what you mean by providing constraints as input, do you mean a "constraints object" as described here? https://fullcalendar.io/docs/eventConstraint

**Comment by sanderdewilde** [comment]
Hi acerix,

Thanks for the reply. This approach will work.

Regarding the constraint, that is indeed what I mean. So in your code, this would come down to:
```js
eventAllow: function(dropInfo, draggedEvent) {
  let constraint = {
    resourceIds: [ 'a', 'b', 'c' ]
  };
  
  let allowed = someFnToCheckIfAllowed(dropInfo, constraint);
  
  if (!allowed) {
    if (!headEl.contains(styleEl)) {
      headEl.appendChild(styleEl);
    }
  }
  else {
    if (headEl.contains(styleEl)) {
      headEl.removeChild(styleEl);
    }
  }
  return true
}
```

So what I meant to ask was, can we reuse some logic from the fullcalendar core so that we wouldn't have to implement `someFnToCheckIfAllowed` ourselves?

Thanks,
Sander
